File: 0088-merge-sorted-array/0088-merge-sorted-array.java
class Solution {
    public void merge(int[] nums1, int m, int[] nums2, int n) {
        int a = nums1.length-1, b=n;
        n=n-1;
        m=m-1;
        while(m>=0 && n>=0){
            if(nums1[m]>=nums2[n]){
                nums1[a]=nums1[m];
                m--;
                a--;
            }
            else{
                nums1[a]=nums2[n];
                n--;
                a--;
            }
        }
        
        while(n>=0){
            nums1[a]=nums2[n];
                     n--;
                a--;
        }
    }
}

File: 0088-merge-sorted-array/0088-merge-sorted-array.cpp
class Solution {
public:
    void merge(vector<int>& nums1, int m, vector<int>& nums2, int n) {
        int left = m-1;
        int right = n-1;
        int end = m+n-1;
        while(right>=0){
            if(left>=0 && nums1[left]>nums2[right]){
                nums1[end]=nums1[left];
                left--;
            }
            else{
                nums1[end]=nums2[right ];
                right--;
            }
            end--;
        }
    }
};